app/assets/javascripts/common/reports.js

```diff
@@ -1237,6 +1246,138 @@ const intersectDataVerticalLine = {
   },
 };
 
+function goalLinePlugin(goalValue) {
+  return {
+    id: "goalLine",
+    beforeDraw: (chart) => {
+      const ctx = chart.ctx;
+      ctx.save();
+      canvasDrawGoalLine(chart, goalValue);
+      canvasDrawGoalTextBubble(chart, goalValue);
+      canvasDrawLineFromGoalToBubble(chart, goalValue);
+    },
+  };
+}
+
+function canvasDrawGoalLine(chart, goalValue) {
+  const ctx = chart.ctx;
+  const chartArea = chart.chartArea;
+  const chartBottom = chartArea.bottom;
+  const chartHeight = chart.chartArea.height;
+  const lineYPosition = chartBottom - (chartHeight / 100) * goalValue;
+
+  ctx.beginPath();
+  ctx.moveTo(chartArea.left + 1, lineYPosition);
+  ctx.lineTo(chartArea.right - 1, lineYPosition);
+  ctx.lineWidth = 3;
+  ctx.lineCap = "round";
+  ctx.strokeStyle = chart.config.data.datasets[0].borderColor;
+  ctx.setLineDash([1, 6]);
+  ctx.stroke();
+  ctx.restore();
+}
+
+function canvasDrawRoundRect(ctx, x, y, width, heightVar, radiusVar, fillStyle) {
+  const radius = radiusVar || 5;
+  const height = heightVar + 4
+  const cornerRadius = radius + 1
+
+  ctx.beginPath();
+  ctx.moveTo(x + cornerRadius, y);
+  ctx.lineTo(x + width - cornerRadius, y);
+  ctx.arcTo(x + width, y, x + width, y + cornerRadius, cornerRadius);
+  ctx.lineTo(x + width, y + height - cornerRadius);
+  ctx.arcTo(
+    x + width,
+    y + height,
+    x + width - cornerRadius,
+    y + height,
+    cornerRadius
+  );
+  ctx.lineTo(x + cornerRadius, y + height);
+  ctx.arcTo(x, y + height, x, y + height - cornerRadius, cornerRadius);
+  ctx.lineTo(x, y + cornerRadius);
+  ctx.arcTo(x, y, x + cornerRadius, y, cornerRadius);
+  ctx.closePath();
+
+  ctx.fillStyle = fillStyle || "grey";
+  ctx.fill();
+}
+
+function canvasDrawGoalTextBubble(chart, goalValue) {
+  // draw text
+  const ctx = chart.ctx;
+  const cornerRadius = 10;
+  const xTextPadding = 7;
+  const yTextPadding = 2;
+
+  const rgbaChartColor = chart.config.data.datasets[0].borderColor;
+  const textColor = rgbaChartColor;
+  const rgbaArray = rgbaChartColor.match(/\d+/g);
+  rgbaArray[3] = 0.15;
+  const fillColor = `rgba(${rgbaArray.join(", ")})`;
```

Comment:
Could extract this to a function that takes a color string and changes the `a` value.

Reply:
done 👍